doc/user-guide/file_clients.rst

```diff
@@ -0,0 +1,87 @@
+File clients
+============
+
+.. note::
+
+    This is a guide on ``file clients``. For API docs
+    see :ref:`api-python-client`.
+
+File clients are used for uploading File products to the cloud. Currently two clients are supported for Amazon S3 and Google Cloud respectively.
+
+The upload process happens in two steps:
+
+* Given a local path, the remote path for storing the file is computed. An absolute local file path of ``/path/to/project/out/data.csv`` gets translated to ``path/to/parent/out/data.csv``. Here, ``parent`` is the parent folder in the bucket to store the files.
+* Upload the file using appropriate cloud APIs.
```

Comment:
this line is probably out of place?

Reply:
> am I right to assume that these docs are mostly gathering docs that we already had in other places? (that's fine, I'm just wondering where this came from and if we should remove other references to prevent duplication)

I picked up the client file creating and pipeline configuration bits from `soopervisor` docs and some example scripts in the projects repo. The remaining details have been added based on points listed in the [original issue](https://github.com/ploomber/ploomber/issues/701). I'm not very if the last section and notes section is present elsewhere in the docs

Reply:
sounds good